frontend/src/metabase/parameters/components/ParameterValuePicker/ListPicker/ListPicker.tsx

```diff
@@ -0,0 +1,65 @@
+import { Select, Loader } from "metabase/ui";
+
+import { PickerIcon } from "../ParameterValuePicker.styled";
+import { handleInputKeyup } from "../util";
+
+import S from "./ListPicker.css";
+
+interface ListPickerProps {
+  value: string;
+  values: string[];
+  onChange: (value: string) => void;
+  onClear: () => void;
+  onSearchChange?: (query: string) => void;
+  onDropdownOpen?: () => void;
+  enableSearch: boolean;
+  isLoading: boolean;
+  noResultsText: string;
+  placeholder: string;
+  errorMessage?: string;
+}
+
+// TODO show "remove" button when typing in search for static list parameters
+export function ListPicker(props: ListPickerProps) {
+  const {
+    value,
+    values,
+    onChange,
+    onClear,
+    onSearchChange,
+    onDropdownOpen,
+    enableSearch,
+    placeholder,
+    noResultsText,
+    isLoading,
+    errorMessage,
+  } = props;
+
+  const icon = isLoading ? (
+    <div data-testid="listpicker-loader">
+      <Loader size="xs" />
+    </div>
+  ) : value ? (
+    <PickerIcon name="close" onClick={onClear} />
+  ) : null;
+
+  return (
+    <Select
+      // This is required until we fix the Select to support maxDropdownHeight
+      classNames={{ dropdown: S.dropdown }}
+      error={errorMessage}
+      value={value}
+      data={values}
+      onChange={onChange}
+      rightSection={icon}
+      placeholder={placeholder}
+      searchable={enableSearch}
+      onKeyUp={handleInputKeyup}
+      nothingFound={noResultsText}
+      onSearchChange={onSearchChange}
+      onDropdownOpen={onDropdownOpen}
+      selectOnBlur
```

Comment:
Well, it's hard to say what is a requirement since this is an improvement of the existing workflow, which is in my opinion bad. I ended up removing this part since it is indeed a bit confusing: it is a _part_ of not allowing a required parameter having no default, which was easy to add, but it is incomplete — so I'll add it later.